WordPress/Classes/Services/PostService.m

```diff
@@ -34,7 +34,7 @@ - (Post *)createPostForBlog:(Blog *)blog {
     post.remoteStatus = AbstractPostRemoteStatusSync;
     PostCategoryService *postCategoryService = [[PostCategoryService alloc] initWithManagedObjectContext:self.managedObjectContext];
     PostCategory *category = [postCategoryService findWithBlogObjectID:blog.objectID andCategoryID:blog.settings.defaultCategoryID];
-    if (category) {
+    if (category && category.categoryID.integerValue != PostCategoryUncategorized) {
```

Comment:
Very, very good find.

One comment though.  I know this is extremely nit-picky, but we can avoid the lookup in CoreData by doing instead:

```swift
    if (blog.settings.defaultCategoryID.integerValue != PostCategoryUncategorized) {
        PostCategory *category = [postCategoryService findWithBlogObjectID:blog.objectID andCategoryID:blog.settings.defaultCategoryID];

        if (category) {
            [post addCategoriesObject:category];
        }
    }
```